NPE in IGenerateTypeCheck.generateTypeCheck with record patterns in switch expression inside lambda

ECJ throws NPE when compiling record patterns inside a switch expression inside a lambda.

Reproducer repo: https://github.com/rdeanmcdonald/ecj-record-pattern-lambda-npe

Environment:
      - ECJ version: 3.46.0 (JDTLS 1.58.0-SNAPSHOT)
      - Java: 25.0.1
      - OS: macOS aarch64

Note: Code compiles fine with javac (Java 25)

JDTLS logs:
```
!ENTRY org.eclipse.core.resources 4 2 2026-05-26 10:06:36.897
!MESSAGE Problems occurred when invoking code from plug-in: "org.eclipse.core.resources".
!STACK 0
java.lang.RuntimeException: Internal Error compiling /jdtls-bug-reproducer/src/main/java/reproducer/RecordPatternInLambdaBug.java
	at org.eclipse.jdt.internal.compiler.ProcessTaskManager.processing(ProcessTaskManager.java:139)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:545)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: java.lang.NullPointerException: Cannot read field "resolvedType" because "expectedTypeRef" is null
	at org.eclipse.jdt.internal.compiler.ast.IGenerateTypeCheck.generateTypeCheck(IGenerateTypeCheck.java:89)
	at org.eclipse.jdt.internal.compiler.ast.TypePattern.generateTypeCheck(TypePattern.java:104)
	at org.eclipse.jdt.internal.compiler.ast.RecordPattern.generateCode(RecordPattern.java:251)
	at org.eclipse.jdt.internal.compiler.ast.CaseStatement.generateCode(CaseStatement.java:337)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement$SwitchTranslator.generateSwitchBlock(SwitchStatement.java:833)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement$SwitchTranslator.generateCode(SwitchStatement.java:898)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement.generateCode(SwitchStatement.java:1082)
	at org.eclipse.jdt.internal.compiler.ast.SwitchExpression.generateCode(SwitchExpression.java:309)
	at org.eclipse.jdt.internal.compiler.ast.ReturnStatement.generateCode(ReturnStatement.java:238)
	at org.eclipse.jdt.internal.compiler.ast.Block.generateCode(Block.java:102)
	at org.eclipse.jdt.internal.compiler.ast.LambdaExpression.generateCode(LambdaExpression.java:1272)
	at org.eclipse.jdt.internal.compiler.ast.LambdaExpression.generateCode(LambdaExpression.java:1216)
	at org.eclipse.jdt.internal.compiler.ClassFile.addSpecialMethods(ClassFile.java:1024)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.generateCode(TypeDeclaration.java:639)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.generateCode(TypeDeclaration.java:705)
	at org.eclipse.jdt.internal.compiler.ast.CompilationUnitDeclaration.generateCode(CompilationUnitDeclaration.java:403)
	at org.eclipse.jdt.internal.compiler.Compiler.process(Compiler.java:822)
	at org.eclipse.jdt.internal.compiler.ProcessTaskManager.processing(ProcessTaskManager.java:135)
	... 5 more

!ENTRY org.eclipse.core.resources 4 2 2026-05-26 10:06:36.994
!MESSAGE Problems occurred when invoking code from plug-in: "org.eclipse.core.resources".
!STACK 0
java.lang.RuntimeException: Internal Error compiling /jdtls-bug-reproducer/src/main/java/reproducer/RecordPatternInLambdaBug.java
	at org.eclipse.jdt.internal.compiler.ProcessTaskManager.processing(ProcessTaskManager.java:139)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:545)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: java.lang.NullPointerException: Cannot read field "resolvedType" because "expectedTypeRef" is null
	at org.eclipse.jdt.internal.compiler.ast.IGenerateTypeCheck.generateTypeCheck(IGenerateTypeCheck.java:89)
	at org.eclipse.jdt.internal.compiler.ast.TypePattern.generateTypeCheck(TypePattern.java:104)
	at org.eclipse.jdt.internal.compiler.ast.RecordPattern.generateCode(RecordPattern.java:251)
	at org.eclipse.jdt.internal.compiler.ast.CaseStatement.generateCode(CaseStatement.java:337)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement$SwitchTranslator.generateSwitchBlock(SwitchStatement.java:833)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement$SwitchTranslator.generateCode(SwitchStatement.java:898)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement.generateCode(SwitchStatement.java:1082)
	at org.eclipse.jdt.internal.compiler.ast.SwitchExpression.generateCode(SwitchExpression.java:309)
	at org.eclipse.jdt.internal.compiler.ast.ReturnStatement.generateCode(ReturnStatement.java:238)
	at org.eclipse.jdt.internal.compiler.ast.Block.generateCode(Block.java:102)
	at org.eclipse.jdt.internal.compiler.ast.LambdaExpression.generateCode(LambdaExpression.java:1272)
	at org.eclipse.jdt.internal.compiler.ast.LambdaExpression.generateCode(LambdaExpression.java:1216)
	at org.eclipse.jdt.internal.compiler.ClassFile.addSpecialMethods(ClassFile.java:1024)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.generateCode(TypeDeclaration.java:639)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.generateCode(TypeDeclaration.java:705)
	at org.eclipse.jdt.internal.compiler.ast.CompilationUnitDeclaration.generateCode(CompilationUnitDeclaration.java:403)
	at org.eclipse.jdt.internal.compiler.Compiler.process(Compiler.java:822)
	at org.eclipse.jdt.internal.compiler.ProcessTaskManager.processing(ProcessTaskManager.java:135)
	... 5 more

!ENTRY org.eclipse.jdt.core.manipulation 4 2 2026-05-26 10:06:37.028
!MESSAGE Problems occurred when invoking code from plug-in: "org.eclipse.jdt.core.manipulation".
!STACK 0
java.lang.NullPointerException: Cannot read field "resolvedType" because "expectedTypeRef" is null
	at org.eclipse.jdt.internal.compiler.ast.IGenerateTypeCheck.generateTypeCheck(IGenerateTypeCheck.java:89)
	at org.eclipse.jdt.internal.compiler.ast.TypePattern.generateTypeCheck(TypePattern.java:104)
	at org.eclipse.jdt.internal.compiler.ast.RecordPattern.generateCode(RecordPattern.java:251)
	at org.eclipse.jdt.internal.compiler.ast.CaseStatement.generateCode(CaseStatement.java:337)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement$SwitchTranslator.generateSwitchBlock(SwitchStatement.java:833)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement$SwitchTranslator.generateCode(SwitchStatement.java:898)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement.generateCode(SwitchStatement.java:1082)
	at org.eclipse.jdt.internal.compiler.ast.SwitchExpression.generateCode(SwitchExpression.java:309)
	at org.eclipse.jdt.internal.compiler.ast.ReturnStatement.generateCode(ReturnStatement.java:238)
	at org.eclipse.jdt.internal.compiler.ast.Block.generateCode(Block.java:102)
	at org.eclipse.jdt.internal.compiler.ast.LambdaExpression.generateCode(LambdaExpression.java:1272)
	at org.eclipse.jdt.internal.compiler.ast.LambdaExpression.generateCode(LambdaExpression.java:1216)
	at org.eclipse.jdt.internal.compiler.ClassFile.addSpecialMethods(ClassFile.java:1024)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.generateCode(TypeDeclaration.java:639)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.generateCode(TypeDeclaration.java:705)
	at org.eclipse.jdt.internal.compiler.ast.CompilationUnitDeclaration.generateCode(CompilationUnitDeclaration.java:403)
	at org.eclipse.jdt.core.dom.CompilationUnitResolver.resolve(CompilationUnitResolver.java:1316)
	at org.eclipse.jdt.core.dom.CompilationUnitResolver.resolve(CompilationUnitResolver.java:830)
	at org.eclipse.jdt.core.dom.CompilationUnitResolver.toCompilationUnit(CompilationUnitResolver.java:1413)
	at org.eclipse.jdt.core.dom.CompilationUnitResolver$ECJCompilationUnitResolver.toCompilationUnit(CompilationUnitResolver.java:109)
	at org.eclipse.jdt.core.dom.ASTParser.internalCreateASTCached(ASTParser.java:1412)
	at org.eclipse.jdt.core.dom.ASTParser.lambda$1(ASTParser.java:1291)
	at org.eclipse.jdt.internal.core.JavaModelManager.cacheZipFiles(JavaModelManager.java:5738)
	at org.eclipse.jdt.core.dom.ASTParser.internalCreateAST(ASTParser.java:1291)
	at org.eclipse.jdt.core.dom.ASTParser.createAST(ASTParser.java:933)
	at org.eclipse.jdt.core.manipulation.CoreASTProvider$1.run(CoreASTProvider.java:294)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.jdt.core.manipulation.CoreASTProvider.createAST(CoreASTProvider.java:286)
	at org.eclipse.jdt.core.manipulation.CoreASTProvider.getAST(CoreASTProvider.java:199)
	at org.eclipse.jdt.ls.core.internal.handlers.DocumentHighlightHandler.documentHighlight(DocumentHighlightHandler.java:56)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$12(JDTLanguageServer.java:760)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:667)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:504)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:511)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1450)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2019)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)

!ENTRY org.eclipse.jdt.core.manipulation 4 0 2026-05-26 10:06:37.029
!MESSAGE Error in JDT Core during AST creation
!STACK 0
java.lang.NullPointerException: Cannot read field "resolvedType" because "expectedTypeRef" is null
	at org.eclipse.jdt.internal.compiler.ast.IGenerateTypeCheck.generateTypeCheck(IGenerateTypeCheck.java:89)
	at org.eclipse.jdt.internal.compiler.ast.TypePattern.generateTypeCheck(TypePattern.java:104)
	at org.eclipse.jdt.internal.compiler.ast.RecordPattern.generateCode(RecordPattern.java:251)
	at org.eclipse.jdt.internal.compiler.ast.CaseStatement.generateCode(CaseStatement.java:337)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement$SwitchTranslator.generateSwitchBlock(SwitchStatement.java:833)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement$SwitchTranslator.generateCode(SwitchStatement.java:898)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement.generateCode(SwitchStatement.java:1082)
	at org.eclipse.jdt.internal.compiler.ast.SwitchExpression.generateCode(SwitchExpression.java:309)
	at org.eclipse.jdt.internal.compiler.ast.ReturnStatement.generateCode(ReturnStatement.java:238)
	at org.eclipse.jdt.internal.compiler.ast.Block.generateCode(Block.java:102)
	at org.eclipse.jdt.internal.compiler.ast.LambdaExpression.generateCode(LambdaExpression.java:1272)
	at org.eclipse.jdt.internal.compiler.ast.LambdaExpression.generateCode(LambdaExpression.java:1216)
	at org.eclipse.jdt.internal.compiler.ClassFile.addSpecialMethods(ClassFile.java:1024)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.generateCode(TypeDeclaration.java:639)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.generateCode(TypeDeclaration.java:705)
	at org.eclipse.jdt.internal.compiler.ast.CompilationUnitDeclaration.generateCode(CompilationUnitDeclaration.java:403)
	at org.eclipse.jdt.core.dom.CompilationUnitResolver.resolve(CompilationUnitResolver.java:1316)
	at org.eclipse.jdt.core.dom.CompilationUnitResolver.resolve(CompilationUnitResolver.java:830)
	at org.eclipse.jdt.core.dom.CompilationUnitResolver.toCompilationUnit(CompilationUnitResolver.java:1413)
	at org.eclipse.jdt.core.dom.CompilationUnitResolver$ECJCompilationUnitResolver.toCompilationUnit(CompilationUnitResolver.java:109)
	at org.eclipse.jdt.core.dom.ASTParser.internalCreateASTCached(ASTParser.java:1412)
	at org.eclipse.jdt.core.dom.ASTParser.lambda$1(ASTParser.java:1291)
	at org.eclipse.jdt.internal.core.JavaModelManager.cacheZipFiles(JavaModelManager.java:5738)
	at org.eclipse.jdt.core.dom.ASTParser.internalCreateAST(ASTParser.java:1291)
	at org.eclipse.jdt.core.dom.ASTParser.createAST(ASTParser.java:933)
	at org.eclipse.jdt.core.manipulation.CoreASTProvider$1.run(CoreASTProvider.java:294)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.jdt.core.manipulation.CoreASTProvider.createAST(CoreASTProvider.java:286)
	at org.eclipse.jdt.core.manipulation.CoreASTProvider.getAST(CoreASTProvider.java:199)
	at org.eclipse.jdt.ls.core.internal.handlers.DocumentHighlightHandler.documentHighlight(DocumentHighlightHandler.java:56)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$12(JDTLanguageServer.java:760)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:667)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:504)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:511)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1450)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2019)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)

!ENTRY org.eclipse.core.resources 4 75 2026-05-26 10:06:37.048
!MESSAGE Errors occurred during the build.
!SUBENTRY 1 org.eclipse.jdt.core 4 75 2026-05-26 10:06:37.048
!MESSAGE Errors running builder 'Java Builder' on project 'jdtls-bug-reproducer'.
!STACK 0
java.lang.RuntimeException: Internal Error compiling /jdtls-bug-reproducer/src/main/java/reproducer/RecordPatternInLambdaBug.java
	at org.eclipse.jdt.internal.compiler.ProcessTaskManager.processing(ProcessTaskManager.java:139)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:545)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: java.lang.NullPointerException: Cannot read field "resolvedType" because "expectedTypeRef" is null
	at org.eclipse.jdt.internal.compiler.ast.IGenerateTypeCheck.generateTypeCheck(IGenerateTypeCheck.java:89)
	at org.eclipse.jdt.internal.compiler.ast.TypePattern.generateTypeCheck(TypePattern.java:104)
	at org.eclipse.jdt.internal.compiler.ast.RecordPattern.generateCode(RecordPattern.java:251)
	at org.eclipse.jdt.internal.compiler.ast.CaseStatement.generateCode(CaseStatement.java:337)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement$SwitchTranslator.generateSwitchBlock(SwitchStatement.java:833)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement$SwitchTranslator.generateCode(SwitchStatement.java:898)
	at org.eclipse.jdt.internal.compiler.ast.SwitchStatement.generateCode(SwitchStatement.java:1082)
	at org.eclipse.jdt.internal.compiler.ast.SwitchExpression.generateCode(SwitchExpression.java:309)
	at org.eclipse.jdt.internal.compiler.ast.ReturnStatement.generateCode(ReturnStatement.java:238)
	at org.eclipse.jdt.internal.compiler.ast.Block.generateCode(Block.java:102)
	at org.eclipse.jdt.internal.compiler.ast.LambdaExpression.generateCode(LambdaExpression.java:1272)
	at org.eclipse.jdt.internal.compiler.ast.LambdaExpression.generateCode(LambdaExpression.java:1216)
	at org.eclipse.jdt.internal.compiler.ClassFile.addSpecialMethods(ClassFile.java:1024)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.generateCode(TypeDeclaration.java:639)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.generateCode(TypeDeclaration.java:705)
	at org.eclipse.jdt.internal.compiler.ast.CompilationUnitDeclaration.generateCode(CompilationUnitDeclaration.java:403)
	at org.eclipse.jdt.internal.compiler.Compiler.process(Compiler.java:822)
	at org.eclipse.jdt.internal.compiler.ProcessTaskManager.processing(ProcessTaskManager.java:135)
	... 5 more
```

--- a/org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/IGenerateTypeCheck.java
+++ b/org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/IGenerateTypeCheck.java
@@ -86,7 +86,12 @@ interface IGenerateTypeCheck {
 				setPatternIsTotalType();
 			}
 			case NO_CONVERSION_ROUTE -> {
-				codeStream.instance_of(expectedTypeRef, expectedTypeRef.resolvedType);
+				if (isUnnamed() && expectedTypeRef == null) { // for a type elided unnamed pattern, there is no need for a type check.
+					consumeProvidedValue(providedType, codeStream);
+					codeStream.iconst_1();
+				} else {
+					codeStream.instance_of(expectedTypeRef, expectedTypeRef.resolvedType);
+				}
 				break;
 			}
 			default -> {
@@ -101,6 +106,11 @@ interface IGenerateTypeCheck {
 	}
 
 	void setPatternIsTotalType();
+
+	/* Overridden in Pattern */
+	default boolean isUnnamed() {
+		return false;
+	}
 
 	default void generateExactConversions(TypeBinding provided, TypeBinding expected, BlockScope scope, CodeStream codeStream) {
 		if (BaseTypeBinding.isExactWidening(expected.id, provided.id)) {

--- a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/SwitchPatternTest.java
+++ b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/SwitchPatternTest.java
@@ -9948,4 +9948,94 @@ public class SwitchPatternTest extends AbstractRegressionTest9 {
 			},
 			"bad");
 	}
+
+	// https://github.com/eclipse-jdt/eclipse.jdt.core/issues/5081
+	// NPE in IGenerateTypeCheck.generateTypeCheck with record patterns in switch expression inside lambda
+	public void testIssue5081() {
+		if (this.complianceLevel < ClassFileConstants.JDK22)
+			return;
+		runConformTest(
+			new String[] {
+				"X.java",
+				"""
+				import java.util.function.Function;
+				import java.util.function.Predicate;
+
+				public class X {
+
+					sealed interface Result<T> permits Valid, ValidWarning, InvalidWarning, InvalidError {}
+					record Valid<T>(T value) implements Result<T> {}
+					record ValidWarning<T>(T value, String warning) implements Result<T> {}
+					record InvalidWarning<T>(String warning) implements Result<T> {}
+					record InvalidError<T>(String error) implements Result<T> {}
+
+					public <Item, Res> Predicate<Item> filter(Function<Item, Result<Res>> validator) {
+						return item -> {
+							var validation = validator.apply(item);
+							return switch (validation) {
+								case Valid<?> _ -> true;
+								case ValidWarning<?>(_, var warning) -> {
+									System.out.println(warning);
+									yield true;
+								}
+								case InvalidWarning<?>(var warning) -> {
+									System.out.println(warning);
+									yield false;
+								}
+								case InvalidError<?>(var error) -> {
+									System.out.println(error);
+									yield false;
+								}
+							};
+						};
+					}
+				}
+				"""
+			});
+	}
+
+	// https://github.com/eclipse-jdt/eclipse.jdt.core/issues/5081
+	// NPE in IGenerateTypeCheck.generateTypeCheck with record patterns in switch expression inside lambda
+	public void testIssue5081_2() {
+		if (this.complianceLevel < ClassFileConstants.JDK22)
+			return;
+		runConformTest(
+			new String[] {
+				"X.java",
+				"""
+				import java.util.function.Function;
+				import java.util.function.Predicate;
+
+				public class X {
+
+					sealed interface Result<T> permits Valid, ValidWarning, InvalidWarning, InvalidError {}
+					record Valid<T>(T value) implements Result<T> {}
+					record ValidWarning<T>(T value, String warning) implements Result<T> {}
+					record InvalidWarning<T>(String warning) implements Result<T> {}
+					record InvalidError<T>(String error) implements Result<T> {}
+
+					public <Item, Res> Predicate<Item> filter(Function<Item, Result<Res>> validator) {
+						return item -> {
+							var validation = validator.apply(item);
+							return switch (validation) {
+								case Valid<?> _ -> true;
+								case ValidWarning<?>(var _, var warning) -> {
+									System.out.println(warning);
+									yield true;
+								}
+								case InvalidWarning<?>(var warning) -> {
+									System.out.println(warning);
+									yield false;
+								}
+								case InvalidError<?>(var error) -> {
+									System.out.println(error);
+									yield false;
+								}
+							};
+						};
+					}
+				}
+				"""
+			});
+	}
 }

